cmd/sops/main.go

```diff
@@ -1511,26 +1517,30 @@ func main() {
 		}
 
 		commandCount := 0
-		if c.Bool("encrypt") {
+		if isDecryptMode {
 			commandCount++
 		}
-		if c.Bool("decrypt") {
+		if isEncryptMode {
 			commandCount++
 		}
-		if c.Bool("rotate") {
+		if isRotateMode {
 			commandCount++
 		}
-		if c.String("set") != "" {
+		if isSetMode {
 			commandCount++
 		}
 		if commandCount > 1 {
 			log.Warn("More than one command (--encrypt, --decrypt, --rotate, --set) has been specified. Only the changes made by the last one will be visible. Note that this behavior is deprecated and will cause an error eventually.")
 		}
 
-		// Load configuration here for backwards compatibility (error out in case of bad config files)
-		_, err = loadConfig(c, fileNameOverride, nil)
-		if err != nil {
-			return toExitError(err)
+		// Load configuration here for backwards compatibility (error out in case of bad config files),
+		// but only when not just decrypting (https://github.com/getsops/sops/issues/868)
+		needsCreationRule := isEncryptMode || isRotateMode || isSetMode || isEditMode
```

Comment:
If I understand correctly, this would only be false if

```
isEncryptMode = false
isRotateMode = false
isSetMode = false
isDecryptMode = true
```

If this is correct, it might be easier to reason about if this is written as 
```suggestion
		// Load configuration here for backwards compatibility (error out in case of bad config files),
		// but only when not just decrypting (https://github.com/getsops/sops/issues/868)
		needsCreationRule := !(isEncryptMode || isRotateMode || isSetMode) && isDecryptMode
```

Reply:
> If I understand correctly, this would only be false if
> 
> ```
> isEncryptMode = false
> isRotateMode = false
> isSetMode = false
> isDecryptMode = true
> ```

Yes, due to the way `isEditMode` works (it's `true` if no other mode is set).

> If this is correct, it might be easier to reason about if this is written as

I find the new expression harder to understand, and the intended behavior - the creation rule needs to be checked here if one of the modes is active that needs it later - less clear.

Also your expression has a boolean error, it evaluates to `true` in case `needsCreationRule` should be `false`, so it would have to be `needsCreationRule := isEncryptMode || isRotateMode || isSetMode || !isDecryptMode`.